eg/Classes/PluginBase.py

```diff
@@ -148,7 +148,15 @@ def AddActionsFromList(self, theList, defaultAction=None):
         self.info.actionGroup.AddActionsFromList(theList, defaultAction)
 
     def AddEvents(self, *eventList):
-        self.info.eventList = eventList
+        if self.info.eventList is None:
+            self.info.eventList = ()
+
+        for event in eventList:
+            if isinstance(event, basestring):
+                event = (event, '')
+            elif isinstance(event, list):
+                event = tuple(event)
+            self.info.eventList += (event,)
```

Comment:
t's funny because they are tied together...

this mechanism does not only apply to the person making the plugin..
you can call it as well when you are making the AddEvent dialog...
have it do what it needs to do all using common code.

so as an example... you use the EventInfo class in the add event dialog..
make this thing so that as items are added it creates the EventInfo instance and stores that instead.
so instead of having to scan the whole tree.... have the EventItem class create an EventInfo class (if it does not already exist) and add it to the eventList.. then all you have to do is load the list with the AddEventDialog. no need to scan anything additional.. because things were added at time of creation.

yes no??

this is the central spot that the EventItem, AddEventsDialog, EventGhostEvent and any things else can access to add items to the list that may not be there. and do any instance checkings and duplicate checking here. so code doesn't have to be duplicated.

it might be easiest to just code up a quick example of what my melon is thinking.. I think it would make the add events dialog code smaller. but it would also allow for adding of items. whether is be a "static" entry or a "dynamic" entry for anywhere inside of EG